stack.py

```diff
@@ -4,20 +4,21 @@ def __init__(self, data, next_item=None):
         self.next_item = next_item
 
     def __str__(self):
-        return self.data
+        return str(self.data)
 
 
-class StackFrame(object):
+class Stack(object):
     def __init__(self, first_item=None):
         self.first_item = first_item
 
-    def push(self, new_item):
-        # push new_item to beginning of list
+    def push(self, val):
+        # push val to beginning of list
+        self.new_item = StackItem(val)
```

Comment:
Why attach `new_item` to self?  Will you ever use it after this method is done running?